### DISCUSSION TITLE
更新日志 - 22.03.6 - 20240524

### DISCUSSION THREAD

**Discussion by jjm2473** [istoreos/istoreos#1597] [Announcements]
# 系统
* 在线升级：
  1. 增加了Cloudflare的CDN，避免高峰时期下载太慢。
    为了让本次固件升级也能通过CDN下载，最近几天会通过重定向方式让老固件使用CDN，如果下载还有问题，可以试试从[SourceForge](https://sourceforge.net/projects/istoreos/files/iStoreOS/)手动下载，或者过几天再试
  2. 修复了“取消下载”功能异常的问题

# 应用
* iStore：升级到0.1.17
  1. iStore维护的IPK仓库增加Cloudflare CDN作为缓存
  2. 默认隐藏架构不兼容的插件（可以在维护页面禁用此功能）

**Comment by Vinke2** [comment]
嗯,很好,很快

**Comment by xiaomeng9597** [comment]
https://fw.koolcenter.com/binary/LinkEase/AutoUpgrade/linkmount/  下载太慢了，云编译直接失败


SHELL= flock /mnt/workdir/openwrt/tmp/.linkmount-binary-1.0.3.tar.gz.flock -c '  	/mnt/workdir/openwrt/scripts/download.pl "/mnt/workdir/openwrt/dl" "linkmount-binary-1.0.3.tar.gz" "b8a5b99751cdd3dd84ac75c94e3cff5b898f910c86f12b4c7ec74dc8b3c510cb" "" "https://fw0.koolcenter.com/binary/LinkEase/AutoUpgrade/linkmount/"    '
+ curl -f --connect-timeout 20 --retry 5 --location https://fw0.koolcenter.com/binary/LinkEase/AutoUpgrade/linkmount/linkmount-binary-1.0.3.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100   114  100   114    0     0    105      0  0:00:01  0:00:01 --:--:--   105

100   167  100   167    0     0    137      0  0:00:01  0:00:01 --:--:--   137

  0     0    0     0    0     0      0      0 --:--:--  0:00:01 --:--:--     0
  0 20.2M    0 32768    0     0  11609      0  0:30:25  0:00:02  0:30:23 33746
  0 20.2M    0 81902    0     0  22577      0  0:15:38  0:00:03  0:15:35 46115
  1 20.2M    1  223k    0     0  48120      0  0:07:20  0:00:04  0:07:16 78675
  1 20.2M    1  288k    0     0  52442      0  0:06:44  0:00:05  0:06:39 78184
  1 20.2M    1  367k    0     0  55011      0  0:06:25  0:00:06  0:06:19 75392
  2 20.2M    2  448k    0     0  59993      0  0:05:53  0:00:07  0:05:46 88286
  2 20.2M    2  448k    0     0  53048      0  0:06:39  0:00:08  0:06:31 75069
  2 20.2M    2  591k    0     0  62831      0  0:05:37  0:00:09  0:05:28 77192
  3 20.2M    3  655k    0     0  63320      0  0:05:34  0:00:10  0:05:24 75582
  3 20.2M    3  703k    0     0  61240      0  0:05:46  0:00:11  0:05:35 69899
  3 20.2M    3  783k    0     0  62954      0  0:05:36  0:00:12  0:05:24 67400
  4 20.2M    4  847k    0     0  63672      0  0:05:32  0:00:13  0:05:19 82073


最后报错 curl: (56) OpenSSL SSL_read: error:0A000126:SSL routines::unexpected eof while reading, errno 0
Download failed.

**Reply by jjm2473** [reply]
再试试，现在重定向到主站了

**Reply by jjm2473** [reply]
云编译的时候最好缓存下dl文件夹，以免重复下载

**Reply by xiaomeng9597** [reply]
> 云编译的时候最好缓存下dl文件夹，以免重复下载

嗯，可以了，不然我还要另外做一个链接😂

**Reply by xiaomeng9597** [reply]
> 再试试，现在重定向到主站了

对了，我反应一个问题啊，ubus程序貌似有问题，https://git.openwrt.org/project/ubus.git  最新提交日期是2023-12-18，现在istoreos用的2022-06-01版本，rk35xx架构，就是3566的盒子ubus老是莫名其妙的挂了，访问web后台直接报502错误，原来的web的uhttpd服务也是会莫名其妙的出问题，重启uhttpd报错 Failed to connect to ubus
Failed to parse json data: unexpected end of data，后面换nginx也一样出这个问题，现在我用脚本直接修改package/system/ubus/Makefile生成文件里面的版本成2023-12-18最新的，然后在编译出来的固件，就不会出现这个问题，希望你们更新一下吧。

相关链接：
https://git.openwrt.org/project/ubus.git

https://github.com/istoreos/istoreos/blob/istoreos-22.03/package/system/ubus/Makefile

**Reply by jjm2473** [reply]
没发现有这些问题，据我所知ubus有个数据大小的限制，应该是不能超过1MB，超过这个大小就会崩溃，官方也没修复。你遇到问题的时候是打开了特定的页面吗？

**Reply by jjm2473** [reply]
> 嗯，可以了，不然我还要另外做一个链接😂

不需要另外做链接，只要把文件放在dl文件夹里就能识别到。

**Reply by xiaomeng9597** [reply]
> 没发现有这些问题，据我所知ubus有个数据大小的限制，应该是不能超过1MB，超过这个大小就会崩溃，官方也没修复。你遇到问题的时候是打开了特定的页面吗？

没有打开什么页面也会的，普通待机运行有时候web服务就会挂掉，后面查找和重启uhttpd或者重启nginx就报ubus服务崩溃了，这种情况只能重启系统才能恢复，现在我还做了一个脚本每一分钟查询nginx有没有挂掉，假如nginx服务挂掉了那百分百是ubus服务崩溃了，只能强制杀死ubus还有rpcd服务，然后在运行这两个服务并且重启nginx，nginx才能正常访问web后台。

**Reply by jjm2473** [reply]
@xiaomeng9597 你到底是用nginx还是uhttpd作为luci的web服务？

**Reply by xiaomeng9597** [reply]
> @xiaomeng9597 你到底是用nginx还是uhttpd作为luci的web服务？

我已经换成nginx了，不过还是存在这个问题，现在我只能写一个sh脚本每分钟查询一次nginx服务运行状态，假如nginx状态是没有运行或者报ubus错误，只能用killall命令强制杀死ubus的进程然后重新启动ubusd和rpcd进程后，istoreos的web luci才能正常访问了

**Reply by xiaomeng9597** [reply]
> @xiaomeng9597 你到底是用nginx还是uhttpd作为luci的web服务？

uhttpd和nginx偶尔都会造成ubus进程崩溃，没办法了

**Reply by jjm2473** [reply]
崩溃以后，看看系统日志有没有ubus，procd，rpcd相关的错误。如果luci没法用，可以终端执行`logread | grep -E 'ubus|prcod|rpcd'`看看

**Comment by JwietPuj-Drin** [comment]
近来微软又要下注 Windows on ARM 了，宝哥考虑做 ARM 架构的 EFI 引导镜像吗？

**Comment by konglongbaba** [comment]
快了百分之九十

**Comment by shenhuash01** [comment]
设置的端口转发没有了，只剩2个，新增一会就会消失

**Comment by OliLi1988** [comment]
真是一头牛
